### PR TITLE
implement channel_builder kind: add_shard

### PR DESCRIPTION
**Pull request by johnfoconnor**
add_shard will publish the event to 1 random channel of N possible sharded channels

this is for load balancing and capacity management and otherwise behaves just like "add"

### CODE REVIEW

**Comment by johnfoconnor** [comment]
i plan to tag an release v2.10 after merging this int master

File: tests/test_channel_builder.py [hunk 1]
```diff
@@ -26,15 +29,20 @@ def test_compose2():
...
     assert 'ody_engine_fps_id' in channels
     assert 'foo' in channels
     assert 'bar' in channels
+    assert 'subscriber_republished_3' in channels
 
 
 def test_some_empty_fields():
+    import random
+
+    random.seed(0)
```
Comment (jhu1997) [inline tests/test_channel_builder.py:44]: I don't know how this test harness works, so I'm going to assume it's not multithreaded and won't mess up your random seed here.

File: sdk/cpp/makefile [hunk 2]
```diff
@@ -30,7 +32,7 @@ endif
 # Default rule does not use python as that requires first time users to have Python3 installed
 #
 brew:
-	mkdir -p build && (cd build ; conan install .. ; conan build .. ; cmake -GNinja -DCMAKE_BUILD_TYPE=MinSizeRel .. ; ninja install)
+	mkdir -p build && (cd build ; conan install --build=missing .. ; conan build .. ; cmake -GNinja -DCMAKE_BUILD_TYPE=MinSizeRel .. ; ninja install)
```
Comment (jhu1997) [inline sdk/cpp/makefile:35]: I'm not familiar with conan so I don't know what this additional argument means, so I'll have to take your word on it.

**Review by jhu1997 — APPROVED** [review]
Looks good, thank you for adding to the unit tests.

File: tests/test_channel_builder.py [hunk 3]
```diff
@@ -26,15 +29,20 @@ def test_compose2():
...
+    assert len(channels) == 5
 
     assert 'sms_republished_v1_neo' in channels
     assert 'ody_engine_fps_id' in channels
     assert 'foo' in channels
     assert 'bar' in channels
+    assert 'subscriber_republished_3' in channels
 
 
 def test_some_empty_fields():
```
Comment (bsergean) [inline tests/test_channel_builder.py:41]: For python code the practice is to put those imports at the top.

File: tests/test_channel_builder.py [hunk 4]
```diff
@@ -109,11 +121,61 @@ def test_remove_channel():
...
+                'id': 'sms_set_rate_control_id',
+                'per_id_counter': 0,
+                'session': '98c3eb2c042e42068fc4d2b39fe720f8',
+                'timestamp': 1582066343026,
+                'version': 1,
+            },
+            'channels': ['a_channel', 'sms_live_shard_v1.wiso.9'],
+        },
+    }
+
```
Comment (bsergean) [inline tests/test_channel_builder.py:166]: nice !

**Review by bsergean — APPROVED** [review]

**Pull request merged**